### PR TITLE
Implement Count Command

### PR DESCRIPTION
**Pull request by case141**
**Added Count Command**
Explanation: Allows users to get the total number of sources.

Changes added:
- CountCommand.java to logic
- CountCommand to parser
- getCount to Model.java and ModelManager.java
- CountCommandTest.java to test file
- CountCommand to test parser

Example: `count`
Result: `Total number of source(s): 0`

*Note: Since the model for Infinity Machine is not completed, this command has been implemented with the current AddressBook4 persons list.

Issue #56

### CODE REVIEW

**Review by suriruhani — DISMISSED** [review]
A useful addition! Approved.

File: src/main/java/seedu/address/logic/commands/CountCommand.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package seedu.address.logic.commands;
+
+import seedu.address.logic.CommandHistory;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+
+public class CountCommand extends Command{
```
Comment (fterh) [inline src/main/java/seedu/address/logic/commands/CountCommand.java]: Travis gonna fail you lol `Command{`

File: src/main/java/seedu/address/logic/commands/CountCommand.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    public static final String COMMAND_WORD = "count";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Counts the number of source(s) in the Infinity Machine. ";
```
Comment (fterh) [inline src/main/java/seedu/address/logic/commands/CountCommand.java]: Should there be a space between `Machine. ";`? If that's the standard then just leave it.

**Comment by fterh** [comment]
Ayy commit titles should be in present imperative tense :D i.e. "Fix travis errors" not "fixed travis errors" but nvm small issue only

**Comment by case141** [comment]
opps I have been careful in the tenses in my previous commits though i missed them in these past 2 commits :(

**Review by ToonDragon — APPROVED** [review]
I approve.

File: src/test/java/seedu/address/logic/commands/AddCommandTest.java [hunk 3]
```diff
@@ -201,6 +201,11 @@ public Person getSelectedPerson() {
         public void setSelectedPerson(Person person) {
             throw new AssertionError("This method should not be called.");
         }
+
+        @Override
+        public int getCount() {
+            return 0;
```
Comment (fterh) [inline src/test/java/seedu/address/logic/commands/AddCommandTest.java:207]: @case141 I overlooked this in my initial code review. This should throw an error (see all the other methods in this class).

Comment (case141) [reply src/test/java/seedu/address/logic/commands/AddCommandTest.java:207]: This was actually automatically added when I was resolving the errors but ok noted.

**Pull request merged**